apps/backend/src/modules/ai-assistant/libs/helpers/initial-chat/generate-scores-response.ts

```diff
@@ -0,0 +1,95 @@
+import { type z } from "zod";
+
+import { ZERO_INDEX } from "~/libs/constants/constants.js";
+import {
+	AIAssistantMessageValidationSchema,
+	type OpenAiResponseMessage,
+} from "~/libs/modules/open-ai/open-ai.js";
+
+import { ChatMessageAuthor, ChatMessageType } from "../../enums/enums.js";
+import {
+	type AIAssistantResponseDto,
+	type ChatMessageDto,
+} from "../../types/types.js";
+import { type balanceAnalysis } from "./balance-analysis.validation-schema.js";
+
+type BalanceAnalysisData = z.infer<typeof balanceAnalysis>;
+
+const generateScoresResponse = (
+	aiResponse: OpenAiResponseMessage,
+): AIAssistantResponseDto | null => {
+	const message = aiResponse.getPaginatedItems().shift();
+
+	if (!message) {
+		return null;
+	}
+
+	const parsedResult = AIAssistantMessageValidationSchema.safeParse(message);
+
+	if (!parsedResult.success) {
+		return null;
+	}
+
+	const contentText: string = parsedResult.data.content[ZERO_INDEX].text.value;
+	const resultData: BalanceAnalysisData = JSON.parse(
+		contentText,
+	) as BalanceAnalysisData;
+
+	let messageIdCounter = ZERO_INDEX;
+
+	const greetingMessage: ChatMessageDto = {
+		author: ChatMessageAuthor.ASSISTANT,
+		createdAt: new Date().toISOString(),
+		id: messageIdCounter++,
+		isRead: false,
+		payload: {
+			text: resultData.messages.greeting,
+		},
+		type: "text",
+	};
+
+	const balanceWheelMessage: ChatMessageDto = {
+		author: ChatMessageAuthor.ASSISTANT,
+		createdAt: new Date().toISOString(),
+		id: messageIdCounter++,
+		isRead: false,
+		payload: {
+			lowestCategories: resultData.lowestCategories.map((category) => {
+				return {
+					categoryId: category.categoryId,
+					categoryName: category.categoryName,
+				};
+			}),
+			text: resultData.messages.comments,
+		},
+		type: "balance wheel",
+	};
+
+	const categoryQuestion = {
+		author: ChatMessageAuthor.ASSISTANT,
+		createdAt: new Date().toISOString(),
+		id: messageIdCounter++,
+		isRead: false,
+		payload: {
+			buttons: [
+				{
+					label: "Yes, 3 lowest",
+					value: "Yes, 3 lowest",
+				},
+				{
+					label: "No, something else",
+					value: "No, something else",
+				},
+			],
+			text: resultData.messages.question,
+		},
+		type: ChatMessageType.QUESTION_WITH_BUTTONS,
+	};
+
+	return {
+		messages: [greetingMessage, balanceWheelMessage, categoryQuestion],
+		threadId: message.thread_id,
+	};
```

Comment:
Why do we have these in Backend? We agreed before, that we won't have them

Reply:
I deleted it. Can add if the frontend decides to have them. 
But I think there is an option where to create chat Messages with pre-setted Answers(buttons) and what should be the trigger to show them.
We have a lot of them in the chat flow, and I understand that we don't have time to implement it.



<img width="311" alt="Screenshot 2024-09-20 at 20 27 22" src="https://github.com/user-attachments/assets/0fc7de1d-bbd0-46ac-a0f5-3f5db737ad68">
<img width="259" alt="Screenshot 2024-09-20 at 20 27 14" src="https://github.com/user-attachments/assets/ffb1044c-6eeb-4fd8-b530-a4e5fe824792">
<img width="323" alt="Screenshot 2024-09-20 at 20 27 02" src="https://github.com/user-attachments/assets/9fe4cd60-737d-4c9f-b570-4bfb727ba325">